File: src/main/java/com/demigodsrpg/chitchat/redis/RedisChatListener.java
package com.demigodsrpg.chitchat.redis;

import com.demigodsrpg.chitchat.RChitchat;
import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.serializer.json.JSONComponentSerializer;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;
import org.redisson.api.listener.MessageListener;

public class RedisChatListener implements MessageListener<String> {
    private final RChitchat R_INST;

    public RedisChatListener(RChitchat rInst) {
        R_INST = rInst;
    }

    @Override
    public void onMessage(CharSequence ignored, String message) {
        if (message != null && !message.startsWith(R_INST.getServerId() + "$")) {
            Component component = JSONComponentSerializer.json().deserialize(message);
            for (Player player : Bukkit.getOnlinePlayers()) {
                player.sendMessage(component);
            }
        }
    }
}

File: src/main/kotlin/com/demigodsrpg/chitchat/redis/RedisChatListener.kt
package com.demigodsrpg.chitchat.redis

import com.demigodsrpg.chitchat.RChitchat
import net.md_5.bungee.api.chat.TextComponent
import net.md_5.bungee.chat.ComponentSerializer
import org.bukkit.Bukkit
import org.redisson.core.MessageListener

class RedisChatListener : MessageListener<String> {
    override fun onMessage(ignored: String, message: String?) {
        if (message != null && !message.startsWith(RChitchat.serverId + "$")) {
            val component = TextComponent(*ComponentSerializer.parse(message))
            for (player in Bukkit.getOnlinePlayers()) {
                player.spigot().sendMessage(component)
            }
        }
    }
}
